### PR TITLE
Feat/lw 6854 portfolio state maintenance investigation

### PR DESCRIPTION
**Pull request by AngelCastilloB**
# Context

We need to conduct an investigate on how input selection affect the user's stake distribution across `n` stake keys during extended operation. 

Given this investigation we should have some idea of the constraints we will face when a user is transacting and delegating to multiple stake pools.

# Proposed Solution

Create a tool that emulates user operation of a wallet over time and tracks several metrics, transaction fees, UTXO set health and drift of delegation portfolio.

multi-delegation-data-gen

### CODE REVIEW

File: packages/e2e/src/tools/multi-delegation-data-gen/example.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      "min": 1000000000,
+      "max": 1000000000
+    },
+    "count": {
+      "min": 1,
+      "max": 1
+    },
+    "period": 1
+  },
+  "inputs": {
```
Comment (mkazlauskas) [inline packages/e2e/src/tools/multi-delegation-data-gen/example.json:19]: This got me confused a little bit, how about:
```suggestion
  "utxoIn": {
    "amount": {
      "min": 1000000000,
      "max": 1000000000
    },
    "count": {
      "min": 1,
      "max": 1
    },
    "period": 1
  },
  "utxoOut": {
```

**Review by mkazlauskas — DISMISSED** [review]
Looking good @AngelCastilloB! One concern is that this is also shipped as a library (`cardano-sdk/e2e` package), and exported (moved) utils don't have direct test coverage. I'd be ok with not having good coverage for this one package, just wanted to point this out so that we can all agree to this.

File: packages/e2e/src/tools/multi-delegation-data-gen/utils/utils.ts [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+  stakeKeyHash: Cardano.RewardAccount.toHash(rewardAccount)
+});
+
+/**
+ * Creates a delegation certificate for the given reward account and pool id.
+ *
+ * @param rewardAccount The reward account that is delegating.
+ * @param poolId The pool id of the pool we are delegating to.
+ */
+const createDelegationCert = (rewardAccount: Cardano.RewardAccount, poolId: Cardano.PoolId): Cardano.Certificate => ({
```
Comment (rhyslbw) [inline packages/e2e/src/tools/multi-delegation-data-gen/utils/utils.ts:66]: This and `createStakeKeyRegistrationCert` seem like they should be part of a more generic utility, maybe within `tx-construction` :thinking:

**Review by rhyslbw — COMMENTED** [review]
Looks good @AngelCastilloB, well done

File: packages/e2e/src/tools/multi-delegation-data-gen/utils/utils.ts (hunk 2 above)
Comment (mirceahasegan) [reply packages/e2e/src/tools/multi-delegation-data-gen/utils/utils.ts:66]: Delegate certificate is [already there](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/tx-construction/src/tx-builder/TxBuilder.ts#LL290C32-L290C32), but it's internal private

**Review by mirceahasegan — APPROVED** [review]
Fantastic work, @AngelCastilloB! :mage_man:

**Review by mirceahasegan — APPROVED** [review]

File: packages/e2e/src/tools/multi-delegation-data-gen/utils/utils.ts (hunk 2 above)
Comment (rhyslbw) [reply packages/e2e/src/tools/multi-delegation-data-gen/utils/utils.ts:66]: Yeah, even better

**Review by mirceahasegan — APPROVED** [review]

**Review by rhyslbw — APPROVED** [review]
Ready for squashing

**Pull request merged**